src/widgets/allergies/allergy-form.component.tsx

```diff
@@ -259,123 +248,122 @@ export default function AllergyForm(props: AllergyFormProps) {
             </div>
-            <div className={`omrs-radio-button ${style.inputMargin}`}>
+            <div className="omrs-radio-button">
               <label>
                 <input
+                  id={AllergyConcepts.ENVIRONMENTAL_ALLERGEN}
                   type="radio"
                   name="allergenType"
-                  value={ENVIRONMENTAL_ALLERGEN_CONCEPT}
+                  value={AllergyConcepts.ENVIRONMENTAL_ALLERGEN}
                   onChange={handleAllergenChange}
                 />
-                <span id="ENVIRONMENTAL">Environmental</span>
+                <span>Environmental</span>
               </label>
             </div>
-            <div className={`omrs-radio-button ${style.inputMargin}`}>
+            <div className="omrs-radio-button">
               <label>
                 <input
+                  id="no-allergies"
                   type="radio"
                   name="allergenType"
                   value="noAllergy"
                   onChange={handleAllergenChange}
                 />
-                <span id="NOALLERGIES">Patient has no allergies</span>
+                <span>Patient has no known allergies</span>
               </label>
             </div>
           </div>
           {allergensArray && (
             <div>
-              <h4 className={`${style.allergyHeader} omrs-bold`}>
-                {getAllergenType(selectedAllergyCategory)
-                  .charAt(0)
-                  .toUpperCase() +
-                  getAllergenType(selectedAllergyCategory)
-                    .slice(1)
-                    .toLowerCase()}{" "}
+              <h4 className={`${styles.allergyHeader} omrs-bold`}>
+                {capitalize(
+                  getAllergyType(selectedAllergyCategory)?.toLowerCase()
+                )}{" "}
                 allergen
               </h4>
-              <div className={style.container}>
-                {allergensArray.map(allergen => (
-                  <div
-                    className={`omrs-radio-button ${style.inputMargin}`}
-                    key={allergen.name.uuid}
-                  >
+              <div className={styles.container}>
+                {allergensArray.map((allergen, index) => (
+                  <div className="omrs-radio-button" key={index}>
                     <label>
                       <input
+                        id={allergen?.uuid}
                         type="radio"
-                        name="codedAllergenUuid"
-                        value={allergen.uuid}
+                        name="allergen-name"
+                        checked={codedAllergenUuid === allergen?.uuid}
+                        value={allergen?.uuid}
                         onChange={evt => setCodedAllergenUuid(evt.target.value)}
+                        required
                       />
-                      <span>{allergen.name.display}</span>
+                      <span>{allergen?.name?.display}</span>
```

Comment:
This is just a feeling, but I don't think we should be using optional chaining quite so liberally. It seems like it might make it harder to notice certain bugs. @FlorianRappl ?